src/Migration/Run/RunService.php

```diff
@@ -177,30 +171,23 @@ public function abortMigration(Context $context): void
         $this->fireTrackingInformation(self::TRACKING_EVENT_MIGRATION_ABORTED, $runId, $context);
     }
 
-    public function cleanupMappingChecksums(string $connectionUuid, Context $context, bool $resetAll = true): void
+    public function startCleanupMappingChecksums(string $connectionId, Context $context): void
     {
-        $sql = <<<SQL
-UPDATE swag_migration_mapping
-SET checksum = null
-WHERE HEX(connection_id) = ?
-AND checksum IS NOT NULL;
-SQL;
-        if ($resetAll === false) {
-            $sql = <<<SQL
-UPDATE swag_migration_mapping AS m
-INNER JOIN swag_migration_data d ON d.mapping_uuid = m.id
-SET m.checksum = null
-WHERE HEX(m.connection_id) = ?
-AND d.written = 0
-AND m.checksum IS NOT NULL;
-SQL;
+        $connection = $this->connectionRepo->search(new Criteria([$connectionId]), $context)->getEntities()->first();
+
+        if ($connection === null) {
+            throw MigrationException::noConnectionFound();
         }
 
         $this->dbalConnection->executeStatement(
-            $sql,
-            [$connectionUuid],
-            [ParameterType::STRING]
+            'UPDATE swag_migration_general_setting SET `is_resetting_checksums` = 1;'
```

Comment:
shouldn't there be a check if it is already resetting check sums before dispatching another message as well? E.g. doing a select query before this update or checking if this update did affect a row (and adding a where clause to only update on =0)